src/App.tsx

```diff
@@ -17,18 +58,37 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setSelectStatus={setSelectStatus}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {todos.length === 0
+              ? (
+                <Loader />
+              ) : (
+                <div className="block">
+                  <TodoList
+                    todos={visibleTodos}
+                    setIsSelected={setIsSelected}
+                    isSelected={isSelected}
+                    setSelectedTodo={setSelectedTodo}
+                  />
+                </div>
+              )}
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isSelected && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          isSelected={isSelected}
+          setIsSelected={setIsSelected}
```

Comment:
try to consider the props renaming guided by naming convention https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364